AsyncDisplayKit/ASControlNode.mm

```diff
@@ -280,30 +281,51 @@ - (void)addTarget:(id)target action:(SEL)action forControlEvents:(ASControlNodeE
     {
       // Do we already have an event table for this control event?
       id<NSCopying> eventKey = _ASControlNodeEventKeyForControlEvent(controlEvent);
-      NSMapTable *eventDispatchTable = _controlEventDispatchTable[eventKey];
+      NSMutableArray *eventTargetActionArray = _controlEventDispatchTable[eventKey];
+      
       // Create it if necessary.
-      if (!eventDispatchTable)
+      if (!eventTargetActionArray)
       {
-        // Create the dispatch table for this event.
-        eventDispatchTable = [NSMapTable weakToStrongObjectsMapTable];
-        if (eventKey) {
-          [_controlEventDispatchTable setObject:eventDispatchTable forKey:eventKey];
+        // Create a target action array for this event.
+        eventTargetActionArray = [[NSMutableArray alloc] init];
+      }
+      else {
+        NSMutableArray *newEventTargetActionArray = [[NSMutableArray alloc] init];
+        NSString *actionString;
+        
+        // UIControl does not support duplicate target-action-events entries, so we replicate that behavior.
+        // See: https://github.com/facebook/AsyncDisplayKit/files/205466/DuplicateActionsTest.playground.zip
+        for (ASControlTargetAction *targetAction in eventTargetActionArray) {
+          BOOL shouldRemove = NO;
+          
+          if ((target == nil && targetAction.target == nil) || targetAction.target == target) {
+            if (!actionString)
+              actionString = NSStringFromSelector(action);
+            
+            if ([NSStringFromSelector(targetAction.action) isEqualToString:actionString])
+              shouldRemove = YES;
+          }
+          
+          if (shouldRemove)
+            continue;
+          
+          [newEventTargetActionArray addObject:targetAction];
```

Comment:
@Adlai-Holler I agree, it's a bit lengthy. Overriding the `isEqual:` is a great idea, although checking the `indexOfObject:` might not work, as it returns an NSUInteger rather than a nullable number, so it's not possible to determine whether the actual target action pair was part of the original set or not. Calling the `containsObject:` then `removeObject:` methods would work but they will check the `isEqual:` twice, so I keep using a foreach cycle and call the `isEquals:` once.